packages/flutter/lib/src/services/platform_channel.dart

```diff
@@ -123,6 +126,79 @@ void _debugRecordDownStream(String channelTypeName, String name,
   _debugLaunchProfilePlatformChannels();
 }
 
+class BackgroundIsolateBinaryMessenger extends BinaryMessenger {
+  final ReceivePort receivePort = ReceivePort();
+  final Map<int, Completer<ByteData?>> _completers = <int, Completer<ByteData?>>{};
+  int _messageCount = 0;
+
+  static BinaryMessenger? _backgroundIsolateBinaryMessenger;
+
+  static void ensureInitialized(ui.RootIsolateToken token) {
+    if (_backgroundIsolateBinaryMessenger == null) {
+      ui.PlatformDispatcher.instance.registerBackgroundIsolate(token);
+      final BackgroundIsolateBinaryMessenger portBinaryMessenger = BackgroundIsolateBinaryMessenger();
+      _backgroundIsolateBinaryMessenger = portBinaryMessenger;
+      portBinaryMessenger.receivePort.listen((dynamic message) {
+        try {
+          final List<dynamic> args = message as List<dynamic>;
+          final int identifier = args[0] as int;
+          final Uint8List bytes = args[1] as Uint8List;
+          final ByteData byteData = ByteData.sublistView(bytes);
+          final Completer<ByteData?> completer =
+              portBinaryMessenger._completers[identifier]!;
+          portBinaryMessenger._completers.remove(identifier);
+          completer.complete(byteData);
+        } catch (exception, stack) {
+          FlutterError.reportError(FlutterErrorDetails(
+            exception: exception,
+            stack: stack,
+            library: 'services library',
+            context:
+                ErrorDescription('during a platform message response callback'),
+          ));
+        }
+      });
+    }
+  }
+
+  static BinaryMessenger? get instance => _backgroundIsolateBinaryMessenger;
+
+  @override
+  Future<void> handlePlatformMessage(String channel, ByteData? data, ui.PlatformMessageResponseCallback? callback) {
+    throw UnimplementedError();
+  }
+
+  @override
+  Future<ByteData?>? send(String channel, ByteData? message) {
+    final Completer<ByteData?> completer = Completer<ByteData?>();
+    final int messageIdentifier = ++_messageCount;
+    _completers[messageIdentifier] = completer;
+    ui.PlatformDispatcher.instance.sendPortPlatformMessage(
+        channel, message, messageIdentifier, receivePort.sendPort);
+    return completer.future;
+  }
+
+  @override
+  void setMessageHandler(String channel, MessageHandler? handler) {
+    throw UnimplementedError();
+  }
+}
+
+BinaryMessenger _findBinaryMessenger() {
+  if (ui.RootIsolateToken.instance == null) {
```

Comment:
I guess this really doesn't matter because nobody but us is ever going to use this API at the end of the day, but if this was a more widely-used API I would definitely suggest having separate "isRootIsolate" vs "RootIsolateToken.instance" affordances, because the former is much clearer than comparing the latter to null.

Reply:
I tried `isRootIsolate` but Dan had some misgivings about talking about "root isolates" in the context of the web.  So now it's "get useBackgroundIsolateBinaryMessenger".